internal/utils/utils.go

```diff
@@ -82,3 +83,9 @@ func ExpandStrings(data interface{}) []string {
 
 	return strSlice
 }
+
+// StorageAddressFormat takes the address in any format and extracts the bus
+// type only (ide/scsi/virtio)
+func StorageAddressFormat(address string) string {
+	return strings.Split(address, ":")[0]
```

Comment:
This can panic (at least with an empty string,  `strings.Split("",":")` returns `[]string{}`), so lets check and handle that case, at least.